File: pophory-domain/src/main/java/com/pophory/pophorydomain/version/infrastructure/ServiceVersionJpaRepository.java
package com.pophory.pophorydomain.version.infrastructure;

import com.pophory.pophorydomain.version.OSType;
import com.pophory.pophorydomain.version.ServiceVersion;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ServiceVersionJpaRepository extends JpaRepository<ServiceVersion, Long> {

    Optional<ServiceVersion> findByOsType(OSType osType);
}

File: pophory-domain/src/main/kotlin/com/pophory/pophorydomain/version/infrastructure/ServiceVersionJpaRepository.kt
package com.pophory.pophorydomain.version.infrastructure

import com.pophory.pophorydomain.version.OSType
import com.pophory.pophorydomain.version.ServiceVersion
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ServiceVersionJpaRepository : JpaRepository<ServiceVersion, Long> {
    fun findByOsType(osType: OSType?): Optional<ServiceVersion>
}